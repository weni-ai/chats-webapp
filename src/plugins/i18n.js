import Vue from 'vue';
import VueI18n from 'vue-i18n';

// eslint-disable-next-line camelcase
import pt_br from '@/locales/pt_br.json';
import en from '@/locales/en.json';
import es from '@/locales/es.json';

import moment from 'moment';

import('moment/locale/pt-br');
import('moment/locale/es');

moment.locale('pt-br');

Vue.use(VueI18n);

const messages = {
  'pt-br': pt_br,
  en,
  es,
};

const i18n = new VueI18n({
  locale: 'pt-br',
  fallbackLocale: 'pt-br',
  messages,
  dateTimeFormats: {
    'pt-br': {
      short: {
        year: 'numeric',
        month: '2-digit',
        day: '2-digit',
      },
      long: {
        month: 'long',
        day: '2-digit',
      },
    },
    en: {
      short: {
        year: 'numeric',
        month: '2-digit',
        day: '2-digit',
      },
      long: {
        month: 'long',
        day: '2-digit',
      },
    },
    es: {
      short: {
        year: 'numeric',
        month: '2-digit',
        day: '2-digit',
      },
      long: {
        month: 'long',
        day: '2-digit',
      },
    },
  },
});

export default i18n;
