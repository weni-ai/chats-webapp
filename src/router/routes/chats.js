import isMobile from 'is-mobile';

import DesktopHome from '@/views/chats/Home/index.vue';
import MobileHome from '@/views/chats/Mobile/MobileHome.vue';

const Home = isMobile() ? MobileHome : DesktopHome;

const routes = [
  {
    path: '/',
    name: 'root',
    beforeEnter: (from, to, next) => {
      if (to.path === '/') {
        if (isMobile()) {
          next({ name: 'orgs', replace: true });
        } else {
          if (from.query.next) {
            const isGeneralSettings = from.query.next === '/settings/chats';
            next({ path: isGeneralSettings ? '/settings' : from.query.next });
          } else next({ name: 'home', replace: true });
        }
      } else next(to.path);
    },
  },
  {
    path: '/:pathMatch(.*)*',
    name: '404',
    redirect: () => {
      return { path: `/` };
    },
  },
  {
    path: '/rooms',
    name: 'home',
    component: Home,
    props: (route) => ({
      newQuickMessage: route.query.newQuickMessage,
    }),
  },
  {
    path: '/chats/:roomId',
    name: 'room',
    component: Home,
    props: (route) => ({
      roomId: route.params.roomId,
    }),
  },
  {
    path: '/discussions/:discussionId',
    name: 'discussion',
    component: Home,
    props: (route) => ({
      discussionId: route.params.discussionId,
    }),
  },
  {
    path: '/closed-chats',
    name: 'closed-rooms',
    component: () => import('@/views/chats/ClosedChats/index.vue'),
    props: (route) => ({
      contactUrn: route.query.contactUrn,
      sector: route.query.sector,
      tags: route.query.tags,
      startDate: route.query.startDate,
      endDate: route.query.endDate,
      from: route.query.from,
    }),
  },
  {
    path: '/closed-chats/:roomId',
    name: 'closed-rooms.selected',
    component: () => import('@/views/chats/ClosedChats/index.vue'),
    props: (route) => ({
      roomId: route.params.roomId,
    }),
  },
];

export default routes;
