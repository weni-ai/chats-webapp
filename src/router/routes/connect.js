import store from '@/store';

const routes = [
  {
    path: '/loginexternal/:token/',
    name: 'external.login',
    component: null,
    beforeEnter: async (to, from, next) => {
      const { token } = to.params;
      await store.dispatch('auth/setToken', token.replace('+', ' '));

      if (to.query.next) {
        next(to.query.next);
      } else {
        next('/');
      }
    },
  },
];

export default routes;
