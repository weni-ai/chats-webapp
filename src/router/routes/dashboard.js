const routes = [
  {
    path: '/dashboard/manager',
    name: 'dashboard.manager',
    component: () => import('@/views/Dashboard/Manager/index.vue'),
  },
  {
    path: '/dashboard/view-mode/:viewedAgent/:oldModule?',
    name: 'dashboard.view-mode',
    component: () => import('@/views/Dashboard/Manager/index.vue'),
  },
];

export default routes;
