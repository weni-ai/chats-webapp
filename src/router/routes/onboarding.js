const routes = [
  {
    path: '/onboarding',
    name: 'onboarding.agent',
    component: () => import('@/views/onboarding/Agent.vue'),
  },
];

export default routes;
