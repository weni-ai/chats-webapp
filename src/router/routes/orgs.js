const routes = [
  {
    path: '/orgs',
    name: 'orgs',
    component: () => import('@/views/chats/Mobile/MobileSelectOrgProject'),
    beforeEnter: async (to, from, next) => {
      if (from.name === 'home') {
        // eslint-disable-next-line no-param-reassign
        to.params.from = 'home';
      }
      next();
    },
  },
  {
    path: '/orgs/:orgUuid/projects',
    name: 'projects',
    component: () => import('@/views/chats/Mobile/MobileSelectOrgProject'),
    props: (route) => ({
      orgUuid: route.params.orgUuid,
    }),
  },
];

export default routes;
