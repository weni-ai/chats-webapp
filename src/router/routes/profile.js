const routes = [
  {
    path: '/profile/preferences',
    name: 'preferences',
    component: () => import('@/views/profile/Preferences.vue'),
  },
];

export default routes;
