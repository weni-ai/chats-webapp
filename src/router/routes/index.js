import orgs from './orgs';
import chats from './chats';
import dashboard from './dashboard';
import profile from './profile';
import settings from './settings';
import connect from './connect';
import onboarding from './onboarding';

export default [
  ...orgs,
  ...chats,
  ...dashboard,
  ...profile,
  ...settings,
  ...connect,
  ...onboarding,
];
