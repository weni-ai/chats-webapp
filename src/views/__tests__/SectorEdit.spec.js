import { mount, flushPromises } from '@vue/test-utils';
import { describe, it, expect, vi } from 'vitest';
import { createRouter, createWebHistory } from 'vue-router';
import routes from '@/router/routes/settings';
import { createTestingPinia } from '@pinia/testing';
import SectorEdit from '@/views/Settings/Sectors/Edit/index.vue';
import i18n from '@/plugins/i18n.js';
import UnnnicSystemPlugin from '@/plugins/UnnnicSystem';

const mockSector1 = {
  id: 1,
  uuid: '123',
  name: 'Sector 1',
  agents: 10,
  contacts: 50,
};

const store = createTestingPinia({
  initialState: {
    config: {
      project: { name: 'Project 1' },
    },
    settings: {
      sectors: [mockSector1],
      currentSector: mockSector1,
      isLoadingSectors: false,
    },
  },
});

const router = createRouter({
  history: createWebHistory(),
  routes,
});

const createWrapper = (props) => {
  return mount(SectorEdit, {
    props,
    global: {
      plugins: [router, store, UnnnicSystemPlugin, i18n],
      stubs: {
        UnnnicTab: true,
      },
      mocks: {
        $router: {
          replace: vi.fn(),
        },
      },
    },
  });
};

describe('EditSector.vue', () => {
  let wrapper;

  beforeEach(async () => {
    router.push('/settings/sectors/1?tab=general');
    wrapper = createWrapper({});
    await flushPromises();
  });

  it('should render SectorEditHeader with the current sector name', async () => {
    const viewHeader = wrapper.findComponent(
      '[data-testid=sector-edit-view-header]',
    );

    expect(viewHeader.props('sectorName')).toBe('Sector 1');
  });

  it('should render UnnnicTab with the correct tabs', () => {
    const unnnicTab = wrapper.findComponent(
      '[data-testid=sector-edit-view-tab-list]',
    );
    expect(unnnicTab.props('tabs')).toEqual(wrapper.vm.tabIds);
  });

  it('should update the active tab and router query on tab change', async () => {
    const unnnicTab = wrapper.findComponent(
      '[data-testid=sector-edit-view-tab-list]',
    );

    const spyRouterReplace = vi.spyOn(wrapper.vm.$router, 'replace');
    await unnnicTab.vm.$emit('change', 'extra_options');

    expect(wrapper.vm.activeTab.id).toBe('extra_options');
    expect(spyRouterReplace).toHaveBeenCalledWith({
      query: { tab: 'extra_options' },
    });
  });

  it('Should match the snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
