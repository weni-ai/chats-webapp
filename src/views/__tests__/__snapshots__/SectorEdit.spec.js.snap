// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditSector.vue > Should match the snapshot 1`] = `
"<section data-v-ac9ddd5b="" class="sector-edit-view">
  <header data-v-573bd910="" data-v-ac9ddd5b="" class="sector-edit-view__header" data-testid="sector-edit-view-header">
    <h1 data-v-573bd910="" class="header__title" data-testid="header-title">Setor Sector 1</h1>
    <p data-v-573bd910="" class="header__description" data-testid="header-description">config_chats.edit_sector.description</p>
  </header>
  <div data-v-b18efda3="" data-v-ac9ddd5b="" class="tab size-md" data-testid="sector-edit-view-tab-list">
    <header data-v-b18efda3="" class="tab-header">
      <ul data-v-b18efda3="" class="tab-content">
        <li data-v-b18efda3="" class="tab-head tab-head--active">sector.general
          <!---->
        </li>
        <li data-v-b18efda3="" class="tab-head">sector.extra_options
          <!---->
        </li>
        <li data-v-b18efda3="" class="tab-head">Filas
          <!---->
        </li>
        <li data-v-b18efda3="" class="tab-head">Mensagens r√°pidas
          <!---->
        </li>
      </ul>
    </header>
    <main data-v-b18efda3="" class="tab-body">
      <div data-v-b18efda3="" class="tab-panel"></div>
    </main>
  </div>
</section>"
`;
